### PR TITLE
Fill out docs

### PR DESCRIPTION
**Pull request by rafaqz**
@lazarusA not finished but at least some text

### CODE REVIEW

File: docs/src/ext_dd.md [hunk 1]
```diff
@@ -1 +1,85 @@
...
+## `dims`
+
+Objects extending DimensionalData.jl that have dimensions must return 
+a `Tuple` of constructed `Dimension`s from `dims(obj)`. 
+
+### `Dimension` axes
+
+Dimensions return from `dims` should hold a `LookupArray` or in some cases 
+just an `AbstractArray` (like wiht `DimIndices`). When attached to 
+mullti-dimensional objects, lookups must be the _same length_ as the axis
```
Comment (felixcremer) [inline docs/src/ext_dd.md:25]:
```suggestion
multi-dimensional objects, lookups must be the _same length_ as the axis 
```

File: docs/src/ext_dd.md [hunk 2]
```diff
@@ -1 +1,85 @@
...
+
+Objects extending DimensionalData.jl that have dimensions must return 
+a `Tuple` of constructed `Dimension`s from `dims(obj)`. 
+
+### `Dimension` axes
+
+Dimensions return from `dims` should hold a `LookupArray` or in some cases 
+just an `AbstractArray` (like wiht `DimIndices`). When attached to 
+mullti-dimensional objects, lookups must be the _same length_ as the axis 
+of the array it represents, and `eachindex(A, i)` and `eachindex(dim)` must
```
Comment (felixcremer) [inline docs/src/ext_dd.md:26]: What are A and dim here? And there seems to be no eachindex(A, i) at least this also fails for normal Arrays.

File: docs/src/ext_dd.md [hunk 3]
```diff
@@ -1 +1,85 @@
...
+```
+
+`rebuild` has keyword versions automatically generated for all objects
+using [ConstructionBase.jl](https://github.com/JuliaObjects/ConstructionBase.jl). 
+
+These will work without further work as long as your object has the fields 
+used by DimensionalData.jl objects. For example, `AbstractDimArray` will 
+receive these keywords in `rebuild`: `data`, `dims`, `refdims`, `name`, `metadata`. 
+
+If your `AbstractDimArray` does not have all these fields, you must implement
```
Comment (felixcremer) [inline docs/src/ext_dd.md:64]: Could I also fake these fields by overloading `getproperty(MyDimArray, :dims)? Or would this be a bad idea?

File: docs/src/ext_dd.md [hunk 4]
```diff
@@ -1 +1,85 @@
...
+
+| Type             | Keywords                                                    | Arguments            |
+|------------------|------------------------------------------------------------ |----------------------|
+| AbstractDimArray | data, dims, [refdims, name, metadata]                       | as with kw, in order |
+| AbstractDimStack | data, dims, [refdims], layerdims, [metadata, layermetadata] | as with kw, in order |
+| Dimension        | val                                                         | val                  |
+| Selector         | val, [atol]                                                 | val                  |
+| LookupArray      | data, [order, span, sampling, metadata]                     | keywords only        |
+
+You can always add your ownd keywords to `rebuild` calls, but these will only
```
Comment (felixcremer) [inline docs/src/ext_dd.md:84]:
```suggestion
You can always add your own keywords to `rebuild` calls, but these will only
```

File: docs/src/integrations.md [hunk 5]
```diff
@@ -1,7 +1,83 @@
...
+All DynamicGrids.jl `Outputs` are `<: AbstractDimArray`, and 
+`AbstractDimArray` are used for auxiliary data to allow temporal 
+synchronisation during simulations. Notably, this all works on GPUs!
+
+## Analysis
+
+### AstroImages.jl
+
+[AstroImages.jl](http://juliaastro.org/dev/modules/AstroImages/)
+Provides tools to load and visualise astromical images.
```
Comment (felixcremer) [inline docs/src/integrations.md:76]:
```suggestion
Provides tools to load and visualise astronomical images.
```

File: docs/src/plots.md [hunk 6]
```diff
@@ -1 +1,42 @@
...
+```
+
+Other plots also work, here we ignore the axis order and instead favour the
+categorical varable for the X axis:
+
+```@example Makie
+Makie.rainclouds(A)
+```
+
+A lot more is planned for Make.jl plots in future!
```
Comment (felixcremer) [inline docs/src/plots.md:42]:
```suggestion
A lot more is planned for Makie.jl plots in future!
```

File: docs/src/selectors.md [hunk 7]
```diff
@@ -1,29 +1,40 @@
-# Selectors
+# Selectors and LookupArrays
 
-Indexing by value in `DimensionalData` is done with [Selectors](@ref).
-IntervalSets.jl is now used for selecting ranges of values (formerly `Between`).
+http://localhost:5173/DimensionalData.jl/reference#lookuparrays
```
Comment (felixcremer) [inline docs/src/selectors.md:3]: Is this going to work with a localhost link? Or should this be replaced by a proper link once the docs are updated?

File: docs/src/selectors.md [hunk 8]
```diff
@@ -1,29 +1,40 @@
...
+| [`Interval{A,B}(a, b)`] | get all indices between `a` and `b`, as `:closed` or `:open`.                | `UnitRange`       |
+| [`Touches(a, b)`]       | like `..` but includes all cells touched by the interval, not just inside it | `UnitRange`       |
+
+Note: `At`, `Near` and `Contains` can wrap either single values or an
+`AbstractArray` of values, to select one index with an `Int` or multiple 
+indices with a `Vector{Int}`.
 
 Selectors find indices in the `LookupArray`, for each dimension. 
+LookupArrays wrap other `AbstractArray` (often `AbstractRange`) but add
+aditional traits to facilitate fast lookups or specifing point or interval
```
Comment (felixcremer) [inline docs/src/selectors.md:29]:
```suggestion
additional traits to facilitate fast lookups or specifying point or interval
```

File: docs/src/selectors.md [hunk 9]
```diff
@@ -1,29 +1,40 @@
...
+behviour.
+
+Some common `LookupArray` that are:
```
Comment (felixcremer) [inline docs/src/selectors.md:32]:
```suggestion
Some common `LookupArray` types are:
```

File: src/Dimensions/primitives.jl [hunk 10]
```diff
@@ -635,6 +635,14 @@ although it will be for `Array`.
 @inline dimstride(dims::DimTuple, n::Int) = prod(map(length, dims)[1:n-1])
 
 
+"""
+    basedims(ds::Tuple)
+    basedims(d::Union{Dimension,Symbol,Type})
+
+Returns `basetypeof(d)()` or a `Tuple` of called on a `Tuple`.
```
Comment (felixcremer) [inline src/Dimensions/primitives.jl:642]:
```suggestion
Returns `basetypeof(d)()` or a `Tuple` if called on a `Tuple`.
```

File: docs/src/stacks.md [hunk 11]
```diff
@@ -1 +1,29 @@
-# Stacks
\ No newline at end of file
+# Stacks
+
+An `AbstractDimStack` represents a collection of `AbstractDimArray`
+layers that share some or all dimensions. For any two layers, a dimension
+of the same name must have the identical lookup - in fact only one is stored
+for all layers to enforce this consistency.
+
+The behaviour is somewhere ebetween a `NamedTuple` and an `AbstractArray`
```
Comment (felixcremer) [inline docs/src/stacks.md:8]:
```suggestion
The behaviour is somewhere between a `NamedTuple` and an `AbstractArray`
```

File: docs/src/stacks.md [hunk 12]
```diff
@@ -1 +1,29 @@
...
+Base functions like `mean`, `maximum`, `reverse` are applied to all layers of the stack.
+
+`broadcast_dims` broadcasts functions over any mix of `AbstractDimStack` and
+`AbstractDimArray` returning a new `AbstractDimStack` with layers the size of
+the largest layer in the broadcast. This will work even if dimension permutation 
+does not match in the objects.
+
+# Performance 
+
+Indexing stack is fast - indexing a single value return a `NamedTuple` from all layers
```
Comment (felixcremer) [inline docs/src/stacks.md:25]:
```suggestion
Indexing a stack is fast - indexing a single value returning a `NamedTuple` from all layers
```

File: docs/src/stacks.md [hunk 13]
```diff
@@ -1 +1,29 @@
...
+usingally, measures in nanoseconds. There are some compilation overheads to this
```
Comment (felixcremer) [inline docs/src/stacks.md:26]:
```suggestion
usually, measures in nanoseconds. There are some compilation overheads to this
```

File: src/LookupArrays/selector.jl [hunk 14]
```diff
@@ -111,6 +113,7 @@ struct At{T,A,R} <: IntSelector{T}
 end
 At(val; atol=nothing, rtol=nothing) = At(val, atol, rtol)
 
+rebvuild(sel::At, val) = At(val, sel.atol, sel.rtol)
```
Comment (felixcremer) [inline src/LookupArrays/selector.jl:116]:
```suggestion
rebuild(sel::At, val) = At(val, sel.atol, sel.rtol)
```

**Review by felixcremer — CHANGES_REQUESTED** [review]
Nice. I mainly found some typos.

**Comment by lazarusA** [comment]
yes, sorry.. now some updates and fixes are being done here: https://github.com/rafaqz/DimensionalData.jl/pull/629

**Comment by rafaqz** [comment]
I can go through these manually later, whish we could just commit them into a new branch or something

**Pull request merged**